src/bin/flamegraph.rs

```diff
@@ -3,24 +3,38 @@ use std::io::{self, BufReader, Read};
 use std::path::PathBuf;
 use structopt::StructOpt;
 
-use inferno::flamegraph::{self, Options};
+use inferno::flamegraph::{self, FuncFrameAttrsMap, Options};
 
 #[derive(Debug, StructOpt)]
 #[structopt(name = "inferno-flamegraph", author = "")]
 struct Opt {
+    /// File containing attributes to use for the SVG frames of particular functions.
+    /// Each line in the file should be a function name followed by a tab,
+    /// then a sequence of tab separated name=value pairs.
+    #[structopt(long = "nameattr")]
+    nameattr_file: Option<PathBuf>,
+
     /// Collapsed perf output files. With no INFILE, or INFILE is -, read STDIN.
     #[structopt(name = "INFILE", parse(from_os_str))]
     infiles: Vec<PathBuf>,
 }
 
 impl Into<Options> for Opt {
     fn into(self) -> Options {
-        Options {}
+        let func_frameattrs = match self.nameattr_file {
+            Some(file) => match FuncFrameAttrsMap::from_file(&file) {
+                Ok(n) => n,
+                Err(e) => panic!("Error reading {}: {:?}", file.display(), e),
```

Comment:
I wonder whether panicking is the right thing to do here... Do you know what the Perl version does?

Reply:
I don't love panicking here either, but the Perl code will `die` if it can't open the file, or if it detects an invalid format.

```perl
        open my $attrfh, $nameattrfile or die "Can't read $nameattrfile: $!\n";
	while (<$attrfh>) {
		chomp;
		my ($funcname, $attrstr) = split /\t/, $_, 2;
		die "Invalid format in $nameattrfile" unless defined $attrstr;
		$nameattr{$funcname} = { map { split /=/, $_, 2 } split /\t/, $attrstr };
	}
```

Reply:
I guess what we really want is something like `TryInto` so that we could deal with the error more nicely in `main`. This seems fine for now though then!